TST: a flaky test on macOS arm64 (RuntimeWarning: invalid value encountered in matmul)

Seen in gh-30160
@mattip reported seeing this independently too

Basically, the test (sometimes) fails as
```
FAILED numpy/linalg/tests/test_linalg.py::TestQR::test_stacked_inputs[float32-outer_size2-size1] - RuntimeWarning: invalid value encountered in matmul
```

I'm able to reproduce this locally (although unreliably), so I'll attempt a bisection.
possibly related to gh-29820

--- a/numpy/_core/src/common/blas_utils.c
+++ b/numpy/_core/src/common/blas_utils.c
@@ -14,8 +14,13 @@
 #if NPY_BLAS_CHECK_FPE_SUPPORT
 /*
  * Static variable to cache runtime check of BLAS FPE support.
+ * Will always be false (ignore all FPE) when accelerate is the compiled backend
  */
+  #if defined(ACCELERATE_NEW_LAPACK)
+static bool blas_supports_fpe = false;
+  #else
 static bool blas_supports_fpe = true;
+  #endif // ACCELERATE_NEW_LAPACK
 
 #endif // NPY_BLAS_CHECK_FPE_SUPPORT
 

